kratos/spatial_containers/search_wrapper.cpp

```diff
@@ -0,0 +1,467 @@
+                rResult.SetDistance(results_distances[index]);
+            }
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TSearchObject, SpatialSearchCommunication TSpatialSearchCommunication>
+void SearchWrapper<TSearchObject, TSpatialSearchCommunication>::LocalSearchNearest(
+    const PointType& rPoint,
+    ResultType& rResult,
+    const int Rank
+    )
+{
+    // If we are using GeometricalObjectBins we can use the optimized search
+    if constexpr (IsGeometricalObjectBins) {
+        rResult = mpSearchObject->SearchNearest(rPoint);
+        rResult.Get().SetRank(Rank);
+    } else { // Using trees
+        // We search if geometrical objects are provided
+        if (mpPointVector) {
+            // Search nearest
+            double distance = 0.0;
+            auto p_point = mpSearchObject->SearchNearestPoint(rPoint, distance);
+
+            // Set the result
+            rResult = ResultType((p_point->pGetObject()).get(), Rank);
+            rResult.SetDistance(distance);
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TSearchObject, SpatialSearchCommunication TSpatialSearchCommunication>
+void SearchWrapper<TSearchObject, TSpatialSearchCommunication>::LocalSearchIsInside(
+    const PointType& rPoint,
+    ResultType& rResult,
+    const int Rank
+    )
+{
+    // If we are using GeometricalObjectBins we can use the optimized search
+    if constexpr (IsGeometricalObjectBins) {
+        rResult = mpSearchObject->SearchIsInside(rPoint);
+        rResult.Get().SetRank(Rank);
+    } else { // Using trees
+        KRATOS_ERROR << "SearchIsInside not compatible with Search trees" << std::endl;
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TSearchObject, SpatialSearchCommunication TSpatialSearchCommunication>
+void SearchWrapper<TSearchObject, TSpatialSearchCommunication>::KeepOnlyClosestResult(ResultContainerVectorType& rResults)
+{
+    auto distance_lambda = [](ResultContainerType& rResult) -> std::vector<double> {
+        return rResult.GetDistances();
+    };
+    KeepOnlyGivenLambdaResult(rResults, distance_lambda);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TSearchObject, SpatialSearchCommunication TSpatialSearchCommunication>
+void SearchWrapper<TSearchObject, TSpatialSearchCommunication>::KeepOnlyLowestRankResult(ResultContainerVectorType& rResults)
```

Comment:
Because in this case it is implemented for `SearchIsInside` and I dind't know which criteria to follow. You say to keep it if local. I can see if this works.